installinator/src/dispatch.rs

```diff
@@ -184,54 +179,121 @@ impl InstallOpts {
                 }
             });
         let progress_handle = progress_reporter.start();
+        let discovery = &self.discover_opts.mechanism;
 
-        let host_phase_2_artifact = fetch_artifact(
-            &host_phase_2_id,
-            &self.discover_opts.mechanism,
-            &log,
-            &event_sender,
-        )
-        .await?;
+        let engine = UpdateEngine::new(&log, event_sender);
+
+        let host_phase_2_id = ArtifactHashId {
+            // TODO: currently we're assuming that wicketd will unpack the host
+            // phase 2 image. We may instead have the installinator do it.
+            // kind: KnownArtifactKind::Host.into(),
```

Comment:
This just moved from above, but I think this line of the comment can be deleted.

Reply:
Done, thanks for catching this.